overlord/snapstate/handlers.go

```diff
@@ -498,17 +517,29 @@ func (m *SnapManager) doMountSnap(t *state.Task, _ *tomb.Tomb) error {
 	}
 	if readInfoErr != nil {
 		if err := m.backend.UndoSetupSnap(snapsup.placeInfo(), snapType, pb); err != nil {
```

Comment:
wondering if it's worth having  a helper  

undoSetupSnap(st, placeInfo, snapType, pb)

for her are below or it's overkill